core/src/main/java/io/questdb/cutlass/line/tcp/SymbolCache.java

```diff
@@ -104,8 +109,8 @@ void of(CairoConfiguration configuration,
         this.txReader = txReader;
         int symCount = safeReadUncommittedSymbolCount(symbolIndexInTxFile, false);
         path.trimTo(plen);
-        symbolMapReader.of(configuration, path, columnName, columnNameTxn, symCount);
-        symbolValueToKeyMap.clear(symCount);
```

Comment:
That's because we don't want to resize the hash map too aggressively. Example: the table has 1+M symbol values, but 90% of them is in the old partitions and only 10% gets used in ILP messages. Before this PR we'd allocate two arrays of `Numbers.ceilPow2(1+M/0.5)` size at the symbol cache initialization time. With this PR, we're resizing the arrays lazily, on demand, which is a better approach for long-lived ILP ingestion streams.

Reply:
My understand is that when writer thread releases table, it calls `SymbolCache.close()`, which still clears the cache and prevents cross-pollination of symbol values. @ideoma could you double check?

Reply:
Our usual implementation of `of()` does not depend that `close()` is already called. Can we follow same pattern here?

Reply:
> My understanding is that when writer thread releases table, it calls `SymbolCache.close()`, which still clears the cache and prevents cross-pollination of symbol values.

Yes, that's the case. Would be nice if someone could double check it though.

Reply:
> Our usual implementation of `of()` does not depend that `close()` is already called. Can we follow same pattern here?

I agree. I've added a `symbolValueToKeyMap.clear()` call as an extra protection measure in 99af1b82ad3b8d12184793af30765400e8ba2025

Reply:
minor: would be good if `ObjIntHashMap.clear()` does not call `Arrays.fill` when `free == capacity`

Reply:
> minor: would be good if `ObjIntHashMap.clear()` does not call `Arrays.fill` when `free == capacity`

Done in 585b2027d17d3e0747f56f4d3f37eda369eb124f